mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/configuration/impl/MongoDBConfiguration.java

```diff
@@ -92,19 +110,18 @@ public MongoClientOptions buildOptions() {
 
 	public List<MongoCredential> buildCredentials() {
 		if ( getUsername() != null ) {
-			if ( MongoCredential.MONGODB_CR_MECHANISM.equals( authenticationMechanism ) ) {
-				return Arrays.asList( MongoCredential.createMongoCRCredential( getUsername(), getDatabaseName(), password() ) );
-			}
-			if ( MongoCredential.GSSAPI_MECHANISM.equals( authenticationMechanism ) ) {
-				return Arrays.asList( MongoCredential.createGSSAPICredential( getUsername() ) );
-			}
-			if ( MongoCredential.MONGODB_X509_MECHANISM.equals( authenticationMechanism ) ) {
-				return Arrays.asList( MongoCredential.createMongoX509Credential( getUsername() ) );
-			}
-			if ( MongoCredential.PLAIN_MECHANISM.equals( authenticationMechanism ) ) {
-				return Arrays.asList( MongoCredential.createPlainCredential( getUsername(), getDatabaseName(), password() ) );
+			switch ( authenticationMechanism ) {
```

Comment:
Not too important, but you might consider to establish a method on the enum itself instead:

    return authenticationMechanism.getCredentials( getUsername(), getDatabaseName(), password() );

It saves you the switch and you don't even need the error message below.